core/native/refresh/src/main/csharp/Services/IconProvider.cs

```diff
@@ -105,17 +105,21 @@ public T GetApplication(Application application, int size)
         public T GetFileIcon(string filename, bool isFolder, bool large, bool isExecutable)
         {
             string key = string.Empty;
+            string fileInfo = filename;
```

Comment:
Changed logic here to be able to extract local file icons correctly, and not pollute the cache with wrong information.
e.g. calling GetFileIcon("C:\A\B.exe") would pollute cache key "ext:exe" to forever use that icon.